agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1268,49 +1261,83 @@ bool backhaul_manager::send_autoconfig_search_message(std::shared_ptr<sRadioInfo
 bool backhaul_manager::send_slave_ap_metric_query_message(uint16_t mid,
                                                           const std::vector<sMacAddr> &bssid_list)
 {
-    bool ret = false;
+    // save mid and clear previous query
+    m_expected_ap_metrics_response.reset_to_new_mid(mid);
+
+    // compare two mac addresses using string compare on their octets
+    auto mac_comp = [](const sMacAddr &addr1, const sMacAddr &addr2) -> bool {
```

Comment:
@arnout ?

Reply:
So my point of view is that it is a bad idea to define a comparison operator on things that are not ordered. mac1 < mac2 has no meaning at all. It would be the same as defining operator+ on MAC addresses. The only thing that defining such an operator does is to motivate people to use std::map or std::set, which are not appropriate containers in most cases.

The only reason why a comparison operation is defined here, is because the standard library developers are too lazy :-) to implement an intersection algorithm that works on unordered containers (probably because that would be highly inefficient). And of course, for this particular case, there is a good reason to use an ordered container, because it makes the intersection operation efficient.

However, if both Ran and Mario have no qualms about defining an operator that has only implementation semantics and no real-world semantics, I'm not going to stop it.

Reply:
We need memcmp() because we are implementing operator<

Reply:
The `memcmp` is in the internal lambda function. I could compare it however you want. Not see how it is related to operator<.

Reply:
Please note that the result of `memcmp()` is compared against 0 with operator<, not with operator==